source/app/schedulers/instance_scheduler.py

```diff
@@ -137,7 +137,16 @@ def _regions(self):
     @property
     def _sts(self):
         if self._sts_client is None:
-            self._sts_client = get_client_with_standard_retry("sts")
+            session = boto3.Session()
+            sts_regional_endpoint = str.format(
+                "https://sts.{}.amazonaws.com", session.region_name
+            )
```

Comment:
Any pros/cons to getting the region name from the session rather than the lambda environment variable?

Reply:
which variable? I've not seen it documented anywhere and so don't trust it to be accurate at all. relying on the executing session seems safer from that perspective

Reply:
That's probably where boto is getting those values from anyway. One "pro" of this is that it's slightly easier to mock it directly in the test so I'll leave it as is.